src/plugins/jartic/datasource.py

```
@@ -0,0 +1,268 @@
+import asyncio
+import logging
+from datetime import datetime
+from decimal import Decimal
+from pathlib import Path
+from typing import Any, AsyncIterator, Dict, List, Optional
```

Ruff (0.12.2)

Import from `collections.abc` instead: `AsyncIterator`

Import from `collections.abc`

(UP035)